### PR TITLE
Remove the demo templates

### PR DESCRIPTION
**Pull request by youknowriad**
This is based on #26414 

It is time to remove the demo templates embedded on the repo, these are not maintained properly and we should better focus on making sure FSE works properly with a companion theme and this theme is going to be 2021-blocks that is being loaded automatically in #26414

### CODE REVIEW

File: packages/e2e-tests/specs/experiments/multi-entity-editing.test.js [hunk 1]
```diff
@@ -152,10 +152,7 @@ describe( 'Multi-entity editor states', () => {
 	const templatePartName = 'Test Template Part Name Edit';
 	const nestedTPName = 'Test Nested Template Part Name Edit';
 
-	useExperimentalFeatures( [
-		'#gutenberg-full-site-editing',
-		'#gutenberg-full-site-editing-demo',
-	] );
+	useExperimentalFeatures( [ '#gutenberg-full-site-editing' ] );
```
Comment (youknowriad) [inline packages/e2e-tests/specs/experiments/multi-entity-editing.test.js:155]: I suspect that the changes here are probably going to impact the e2e tests, we should rework those to rely on 2021-blocks instead. (active the theme at the beginning and disable it in the end)

**Comment by mtias** [comment]
Yes, please!

File: packages/e2e-tests/specs/experiments/multi-entity-editing.test.js (hunk 1 above)
Comment (Addison-Stavlo) [reply packages/e2e-tests/specs/experiments/multi-entity-editing.test.js:155]: > we should rework those to rely on 2021-blocks instead. (active the theme at the beginning and disable it in the end)

That makes sense to me.  I would expect the tests would still work after we ensure that theme is enabled.

Comment (Addison-Stavlo) [reply packages/e2e-tests/specs/experiments/multi-entity-editing.test.js:155]: If 2021-blocks is loaded automatically in wp-env config, wouldn't:

> active the theme at the beginning and disable it in the end

be unnecessary for the test?  Or is it common that other themes are activated/deactivated in other e2es?

Comment (youknowriad) [reply packages/e2e-tests/specs/experiments/multi-entity-editing.test.js:155]: Regardless of what theme we enable by default, we might have tests that rely on regular themes to be enabled (like today), so we can try enabling it by default and see.

**Comment by david-szabo97** [comment]
Code looks good so far. PHP unit tests are failing which can be fixed quickly with a `composer run format`.

**Review by vindl — APPROVED** [review]
This looks good to me! 🚀 

Should we update the message [next to FSE experiment](https://github.com/WordPress/gutenberg/blob/d7cea07707aa766c869c511b3f19a6890cbbba32/lib/experiments-page.php#L61) to note that compatible theme is required for it to work?

**Comment by youknowriad** [comment]
The performance tests for site-editor are failing here, problem here I can't reproduce so I'm not sure what's happening. Any idea @ockham

**Comment by ockham** [comment]
> The performance tests for site-editor are failing here, problem here I can't reproduce so I'm not sure what's happening. Any idea @ockham

The performance tests rely on the demo templates to be present; the selector that's failing is looking for the `.wp-block` class.
I'm not sure how the Twenty Twenty One theme is replacing the demot template; I guess we need to make sure it's present and active for the perf test.

To repro locally, I guess it's best to start from scratch (i.e. either destroy your local wp-env, or even start with a fresh git clone).

**Comment by ockham** [comment]
> I'm not sure how the Twenty Twenty One theme is replacing the demot template; I guess we need to make sure it's present and active for the perf test.

Ah, just noticed you're activating it here: https://github.com/WordPress/gutenberg/pull/26419/files#diff-dac6b5792388c0d2844f5ec7201465c2a1d35b2bc18c580e0aa811a812af2f17R30

Maybe the theme isn't getting properly installed in the GH action?

**Comment by ockham** [comment]
This looks relevant: https://github.com/WordPress/gutenberg/blob/83f08f70a4a7aa612150146a5d3e4bd26066e963/package.json#L248

Performance (and e2e) tests are still `wp-scripts` (rather than `wp-env` based). So the [addition](https://github.com/WordPress/gutenberg/pull/26414/files#diff-9fd21b09a0144355cfd001e3952767df08b2e04241994470aa1d627f797b171eR6) of the Twenty Twenty One Blocks theme to `.wp-env.json` probably doesn't have any effect for `wp-scripts` based stuff like this.

**Comment by youknowriad** [comment]
> Performance (and e2e) tests are still wp-scripts (rather than wp-env based). So the addition of the Twenty Twenty One Blocks theme to .wp-env.json probably doesn't have any effect for wp-scripts based stuff like this.

wp-scripts is used to run the test, wp-env to setup the environment and both are used here. I think the failure would have been different if the theme was not available.

**Comment by noahtallen** [comment]
> wp-scripts is used to run the test, wp-env to setup the environment and both are used here

yep. wp-scripts just says "run the e2e suite on $some_url", and it just so happens that $some_url is the wp-env WordPress instance

**Comment by youknowriad** [comment]
Looks like, the screen is taking longer to load, that's it 😬

File: lib/experiments-page.php [hunk 2]
```diff
@@ -58,21 +58,10 @@ function gutenberg_initialize_experiments_settings() {
 		'gutenberg-experiments',
 		'gutenberg_experiments_section',
 		array(
-			'label' => __( 'Enable Full Site Editing (Warning: this will replace your theme and cause potentially irreversible changes to your site. We recommend using this only in a development environment.)', 'gutenberg' ),
+			'label' => __( 'Enable Full Site Editing (Warning: A block-based theme is required for this experiment to work properly. We recommend using this only in a development environment.)', 'gutenberg' ),
```
Comment (gziolo) [inline lib/experiments-page.php:61]: That's helpful 👍

**Comment by ockham** [comment]
> > wp-scripts is used to run the test, wp-env to setup the environment and both are used here
> 
> yep. wp-scripts just says "run the e2e suite on $some_url", and it just so happens that $some_url is the wp-env WordPress instance

Ah, TIL! Haven't looked deeply into `wp-scripts`, thanks for clearing that up :+1:

File: packages/e2e-tests/specs/performance/site-editor.test.js [hunk 3]
```diff
@@ -52,7 +55,7 @@ describe( 'Site Editor Performance', () => {
 		while ( i-- ) {
 			const startTime = new Date();
 			await page.reload();
-			await page.waitForSelector( '.wp-block' );
+			await page.waitForSelector( '.wp-block', { timeout: 120000 } );
```
Comment (ellatrix) [inline packages/e2e-tests/specs/performance/site-editor.test.js:58]: Why is such a large timeout needed?

Comment (ellatrix) [reply packages/e2e-tests/specs/performance/site-editor.test.js:58]: I see the comment that the screen is taking longer. I'm surprised this takes longer than 30 seconds. :)

**Pull request merged**